exonum/src/blockchain/service.rs

```diff
@@ -32,128 +32,177 @@ use node::{ApiSender, Node, State, TransactionSend};
 use blockchain::{Blockchain, ConsensusConfig, Schema, StoredConfiguration, ValidatorKeys};
 use helpers::{Height, Milliseconds, ValidatorId};
 
-/// A trait that describes transaction processing rules (a group of sequential operations
-/// with the Exonum storage) for the given `Message`.
+/// Transaction processing functionality for `Message`s allowing to apply authenticated, atomic,
+/// constraint-preserving groups of changes to the blockchain storage.
+///
+/// See also [the documentation page on transactions][doc:transactions].
+///
+/// [doc:transactions]: https://exonum.com/doc/architecture/transactions/
 pub trait Transaction: Message + ExonumJson + 'static {
-    /// Verifies the transaction, which includes the message signature verification and other
-    /// specific internal constraints. verify is intended to check the internal consistency of
-    /// a transaction; it has no access to the blockchain state.
-    /// If a transaction fails verify, it is considered incorrect and cannot be included into
+    /// Verifies the internal consistency of the transaction. `verify` should usually include
+    /// checking the message signature (via [`verify_signature`]) and, possibly,
+    /// other internal constraints. `verify` has no access to the blockchain state;
+    /// checks involving the blockchains state must be preformed in [`execute`](#tymethod.execute).
+    ///
+    /// If a transaction fails `verify`, it is considered incorrect and cannot be included into
     /// any correct block proposal. Incorrect transactions are never included into the blockchain.
     ///
     /// *This method should not use external data, that is, it must be a pure function.*
+    ///
+    /// [`verify_signature`]: ../messages/trait.Message.html#method.verify_signature
```

Comment:
FYI, the nightly Rust recently gained ability to refer to other items via paths like `exonum::messages::Message::verify_signature`, so we'll be able to get rid of this hand-linking some time in the future.